core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/integration_test.go

```diff
@@ -63,12 +62,10 @@ func TestIntegration_LogEventProvider(t *testing.T) {
 	ids, addrs, contracts := deployUpkeepCounter(ctx, t, n, ethClient, backend, carrol, logProvider)
 	lp.PollAndSaveLogs(ctx, int64(n))
 
-	go func() {
-		if err := logProvider.Start(ctx); err != nil {
-			t.Logf("error starting log provider: %s", err)
-			t.Fail()
-		}
-	}()
+	if err := logProvider.Start(ctx); err != nil {
+		t.Logf("error starting log provider: %s", err)
+		t.Fail()
```

Comment:
These don't need to be called from separate goroutines, and it was actually a problem since the call to `Fail` could race with the end of the test.